hdf4: enable fortran and netcdf support

sheepforce opened the pull request:
###### Description of changes

Some exotic quantum mechanics packages require the fortran and NetCDF support of HDF4 and I think there is no harm in enabling them by default. This PR does exactly this.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

sheepforce commented on pkgs/tools/misc/hdf4/default.nix:
```diff
@@ -50,15 +52,16 @@ stdenv.mkDerivation rec {
   ++ lib.optional javaSupport jdk
   ++ lib.optional szipSupport szip
-  ++ lib.optional uselibtirpc libtirpc;
+  ++ lib.optional uselibtirpc libtirpc
+  ;
```
This is the formatting required by nixpkgs-fmt. I would rather keep it like this so that the formatter can actually do its job :)

sheepforce replied:
Interesting, you are right an I am too ^^ It actually accepts both. Have moved it now.

markuskowa commented on pkgs/tools/misc/hdf4/default.nix:
```diff
@@ -12,6 +12,8 @@
 , szip
 , javaSupport ? false
 , jdk
+, fortranSupport ? false
```
This can actually be `true` per default. The problem is not the Fortran support but the NetCDF support, which breaks `gdal`.

sheepforce replied:
Ah you are right, I've messed it up. Is fixed.

sheepforce replied:
OK, defaults to false again. I am surprised how fragile this all is ^^

markuskowa commented:
@GrahamcOfBorg build gdal gnudatalanguage

sheepforce committed:
hdf4: enable fortran and netcdf support


hdf4: fortran disabled by default


hdf4: fix formatting


hdf4: optional netcdf support


hdf4: disable fortran by default
